exporter/postgres/setup_pg10.sql

```diff
@@ -10,6 +10,29 @@ GRANT pg_monitor to ccp_monitoring;
 
 CREATE SCHEMA IF NOT EXISTS monitor AUTHORIZATION ccp_monitoring;
 
--- There are currently no SECURITY DEFINER functions required as long as the "pg_monitor" role is granted to ccp_monitoring.
+CREATE OR REPLACE FUNCTION monitor.pgbackrest_info() returns jsonb 
```

Comment:
Permanent table, or some other way to store the data for use in the function is fine.

If the info command breaks, that causes errors in the postgres logs themselves and nothing is returned by the exporter. Would have to think about how to handle that here.

Reply:
However, if the repo is unavailable for any reason, that will cause the archive_command to fail. That's one of the other new metrics I added in, so that alert would go off.

Reply:
Updated the setup_pg11.sql file to create a table to use and updated the function to use that. Also found out that the COPY PROGRAM command doesn't actually return an error if the command it calls fails. It just copies in 0 rows. So accounting for the table being empty and throwing an exception accordingly.

Unfortunately from the Prometheus side, when the check fails, the metric completely disappears from the exporter output. This means the alert never fires. The only way around this is to set up an absence alert. And the "job" names must be known in order to create absence alerts, so we really can't have these active by default. We can just provide an example like I did here

https://github.com/CrunchyData/pgmonitor/blob/master/prometheus/crunchy-alert-rules.yml#L60.

Reply:
may be you add a timestamp couple and stale info can be treated as failed `pgbackrest info` command

Reply:
That seems to over complicate things. And would just be another table that's then constantly growing and needs maintenance by the user.

I've added an example absence alert for if the backrest command itself is failing. Also, while testing this, having the function now throw an explicit exception when an error is encountered causes the `PGExporterScrapeError` alert to fire as well, indicating there's a problem with the exporter collecting metrics. So by default an alert will now be thrown when the backrest command is failing, it just won't be as clear as having an explicit absence alert for the backups.